### PR TITLE
fix(platform): compare target refs before updating

### PR DESCRIPTION
**Pull request by RahulGautamSingh**
## Changes
- Before updating the target branch of existing PRs check whether the config.baseBranch is diff from PR target branch

## Context
- https://github.com/renovatebot/renovate/discussions/23091

## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [ ] No documentation update is required

## How I've tested my work (please select one)

I have verified these changes via:

- [ ] Code inspection only, or
- [ ] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [x] Both unit tests + ran on a real repository

### CODE REVIEW

**Review by rarkins — COMMENTED** [review]
How does this help?

**Comment by rarkins** [comment]
I'm not seeing a logic change

**Comment by RahulGautamSingh** [comment]
The issue is that Azure throws as error when we try to update targetBranch to same as current. So placing a check to make sure we only update targetBranch when it is different, thus preventing the error.

**Comment by RahulGautamSingh** [comment]
> I expected this to require a change of logic in workers

I thought the same, and then coded it. After that I realised that the checks in the platform files is needed regardless of change in targetBranch so that we check targetBranch != undefined before passing value to the api call. 

So I decidedto make changes to platform files instead of workers and plaftorm files.

**Comment by rarkins** [comment]
Are you saying that `targetBranch` won't be passed to the platform if it's unchanged? Because that doesn't sound right to me - I thought your code passed it every time

**Comment by RahulGautamSingh** [comment]
Correct, workers change is needed.

File: lib/modules/platform/bitbucket-server/index.ts [hunk 1]
```diff
@@ -864,26 +864,26 @@ export async function updatePr({
       throw Object.assign(new Error(REPOSITORY_NOT_FOUND), { statusCode: 404 });
     }
 
+    const body: HttpOptions['body'] = {
```
Comment (viceice) [inline lib/modules/platform/bitbucket-server/index.ts:867]:
```suggestion
    const body: any = {
```
I'm pretty sure that the same as any. we should create an interface for this

File: lib/modules/platform/bitbucket/index.ts [hunk 2]
```diff
@@ -945,20 +949,22 @@ export async function updatePr({
   ).body;
 
   try {
+    const body: BitbucketHttpOptions['body'] = {
```
Comment (viceice) [inline lib/modules/platform/bitbucket/index.ts:952]: same here

File: lib/modules/platform/gitlab/index.ts [hunk 3]
```diff
@@ -712,16 +712,19 @@ export async function updatePr({
     ['open']: 'reopen',
     // TODO: null check (#7154)
   }[state!];
+
+  const body: GitlabHttpOptions['body'] = {
```
Comment (viceice) [inline lib/modules/platform/gitlab/index.ts:716]: same

**Review by rarkins — APPROVED** [review]

**Review by viceice — APPROVED** [review]

**Pull request merged**